src/org/omegat/core/data/ProjectTMXEntry.java

```diff
@@ -0,0 +1,120 @@
+/**************************************************************************
+ OmegaT - Computer Assisted Translation (CAT) tool
+          with fuzzy matching, translation memory, keyword search,
+          glossaries, and translation leveraging into updated projects.
+
+ Copyright (C) 2010 Alex Buloichik
+               2012 Guido Leenders, Thomas Cordonnier
+               2013 Aaron Madlon-Kay
+               2014 Alex Buloichik, Aaron Madlon-Kay
+               Home page: http://www.omegat.org/
+               Support center: https://omegat.org/support
+
+ This file is part of OmegaT.
+
+ OmegaT is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ OmegaT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ **************************************************************************/
+
+package org.omegat.core.data;
+
+import java.util.List;
+import java.util.Objects;
+
+import org.omegat.util.TMXProp;
+
+/**
+ * Storage for TMX entry for project memory
+ *
+ * Variables in this class can be changed only before store to ProjectTMX. After that, all values must be
+ * unchangeable.
+ *
+ * Only RealProject can create and change TMXEntry objects.
+ *
+ * @author Alex Buloichik (alex73mail@gmail.com)
+ * @author Guido Leenders
+ * @author Aaron Madlon-Kay
+ */
+public class ProjectTMXEntry extends TMXEntry {
+    public enum ExternalLinked {
+        // declares how this entry linked to external TMX in the tm/auto/
+        xICE, x100PC, xAUTO, xENFORCED
+    };
+
+    public final boolean defaultTranslation;
+    public final ExternalLinked linked;
+
+    ProjectTMXEntry(ITMXEntry from, boolean defaultTranslation, ExternalLinked linked) {
+        super(from);
+
+        this.defaultTranslation = defaultTranslation;
+        this.linked = linked;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (! super.equals(obj)) {
```

Comment:
Many checkstyle warnings to be fixed to fix based on coding guide on OmegaT.
```
[ant:checkstyle] [WARN] /home/vsts/work/1/s/src/org/omegat/core/data/ProjectTMXEntry.java:64:5: Class 'ProjectTMXEntry' looks like designed for extension (can be subclassed), but the method 'equals' does not have javadoc that explains how to do that safely. If class is not designed for extension consider making the class 'ProjectTMXEntry' final or making the method 'equals' static/final/abstract/empty, or adding allowed annotation for the method. [DesignForExtension]
[ant:checkstyle] [WARN] /home/vsts/work/1/s/src/org/omegat/core/data/ProjectTMXEntry.java:66:14: '!' is followed by whitespace. [NoWhitespaceAfter]
[ant:checkstyle] [WARN] /home/vsts/work/1/s/src/org/omegat/core/data/ProjectTMXEntry.java:80:5: Class 'ProjectTMXEntry' looks like designed for extension (can be subclassed), but the method 'hashCode' does not have javadoc that explains how to do that safely. If class is not designed for extension consider making the class 'ProjectTMXEntry' final or making the method 'hashCode' static/final/abstract/empty, or adding allowed annotation for the method. [DesignForExtension]
[ant:checkstyle] [WARN] /home/vsts/work/1/s/src/org/omegat/core/data/ProjectTMXEntry.java:82:42: '1000' is a magic number. [MagicNumber]
[ant:checkstyle] [WARN] /home/vsts/work/1/s/src/org/omegat/core/data/ProjectTMXEntry.java:82:63: '1000' is a magic number. [MagicNumber]
[ant:checkstyle] [WARN] /home/vsts/work/1/s/src/org/omegat/core/data/ProjectTMXEntry.java:91:14: '!' is followed by whitespace. [NoWhitespaceAfter]
[ant:checkstyle] [WARN] /home/vsts/work/1/s/src/org/omegat/core/data/ProjectTMXEntry.java:104: Line has trailing spaces. [RegexpSingleline]
[ant:checkstyle] [WARN] /home/vsts/work/1/s/src/org/omegat/core/data/ProjectTMXEntry.java:105:5: Class 'ProjectTMXEntry' looks like designed for extension (can be subclassed), but the method 'hasProperties' does not have javadoc that explains how to do that safely. If class is not designed for extension consider making the class 'ProjectTMXEntry' final or making the method 'hasProperties' static/final/abstract/empty, or adding allowed annotation for the method. [DesignForExtension]
[ant:checkstyle] [WARN] /home/vsts/work/1/s/src/org/omegat/core/data/ProjectTMXEntry.java:108: Line has trailing spaces. [RegexpSingleline]
[ant:checkstyle] [WARN] /home/vsts/work/1/s/src/org/omegat/core/data/ProjectTMXEntry.java:109:5: Class 'ProjectTMXEntry' looks like designed for extension (can be subclassed), but the method 'getPropValue' does not have javadoc that explains how to do that safely. If class is not designed for extension consider making the class 'ProjectTMXEntry' final or making the method 'getPropValue' static/final/abstract/empty, or adding allowed annotation for the method. [DesignForExtension]
[ant:checkstyle] [WARN] /home/vsts/work/1/s/src/org/omegat/core/data/ProjectTMXEntry.java:112: Line has trailing spaces. [RegexpSingleline]
[ant:checkstyle] [WARN] /home/vsts/work/1/s/src/org/omegat/core/data/ProjectTMXEntry.java:113:5: Class 'ProjectTMXEntry' looks like designed for extension (can be subclassed), but the method 'hasPropValue' does not have javadoc that explains how to do that safely. If class is not designed for extension consider making the class 'ProjectTMXEntry' final or making the method 'hasPropValue' static/final/abstract/empty, or adding allowed annotation for the method. [DesignForExtension]
[ant:checkstyle] [WARN] /home/vsts/work/1/s/src/org/omegat/core/data/ProjectTMXEntry.java:116: Line has trailing spaces. [RegexpSingleline]
[ant:checkstyle] [WARN] /home/vsts/work/1/s/src/org/omegat/core/data/ProjectTMXEntry.java:117:5: Class 'ProjectTMXEntry' looks like designed for extension (can be subclassed), but the method 'getProperties' does not have javadoc that explains how to do that safely. If class is not designed for extension consider making the class 'ProjectTMXEntry' final or making the method 'getProperties' static/final/abstract/empty, or adding allowed annotation for the method. [DesignForExtension]
```

Reply:
I converted the PR to draft so that we stop concentrate on checkstyle; I have still some points to test before considering everything is correct, and come back to the style before moving again to a real pull request

Reply:
Definition of ProjectTMXEntry moved to another PR